frontend/src/components/TrashPageLoggedinView.tsx

```diff
@@ -0,0 +1,121 @@
+  const [deletedNotesLoading, setDeletedNotesLoading] = useState(true);
+  const [showNotesLoadingError, setShowNotesLoadingError] = useState(false);
+
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [noteToDelete, setNoteToDelete] = useState<DeletedNoteModel | null>(
+    null
+  );
+
+  const { decrement } = useTrashCountStore();
+
+  useEffect(() => {
+    async function loadNotes() {
+      try {
+        setShowNotesLoadingError(false);
+        setDeletedNotesLoading(true);
+        const deletedNotes = await NotesApi.fetcDeletedNotes();
+        setDeletedNotes(deletedNotes);
+      } catch (error) {
+        console.error(error);
+        alert(error);
+        setShowNotesLoadingError(true);
+      } finally {
+        setDeletedNotesLoading(false);
+      }
+    }
+    loadNotes();
+  }, []);
+
+  async function permaDeleteNote() {
+    if (!noteToDelete) return;
+    try {
+      setDeletedNotes(
+        deletedNotes.filter(
+          (existingNote) => existingNote._id !== noteToDelete._id
+        )
+      );
+      decrement();
+      await NotesApi.permaDeleteNote(noteToDelete._id);
+    } catch (error) {
+      console.error(error);
+      alert(error);
+    } finally {
+      setShowDeleteModal(false);
+      setNoteToDelete(null);
+    }
+  }
+
+  async function restoreDeletedNote(note: DeletedNoteModel) {
+    try {
+      setDeletedNotes(
+        deletedNotes.filter((existingNote) => existingNote._id !== note._id)
+      );
+      decrement();
+      await NotesApi.restoreDeletedNote(note._id);
+    } catch (error) {
+      console.error(error);
+      alert(error);
+    }
+  }
+
+  const handleDeleteClick = (note: DeletedNoteModel) => {
+    setNoteToDelete(note);
+    setShowDeleteModal(true);
+  };
+
+  const notesGrid = (
+    <Row xs={1} md={2} xl={3} className={`g-4 ${styles.notesGrid}`}>
+      {deletedNotes.map((note) => (
+        <Col key={note._id}>
+          <DeletedNote
+            note={note}
+            className={styles.note}
+            onDeleteNoteClicked={handleDeleteClick}
+            onRestoreNoteClicked={restoreDeletedNote}
+          />
+        </Col>
+      ))}
+    </Row>
+  );
+
+  return (
+    <>
+      {deletedNotesLoading && <Spinner animation="border" variant="primary" />}
+      {showNotesLoadingError && (
+        <p>Something went wrong. Please refresh the page</p>
+      )}
+      {!deletedNotesLoading && !showNotesLoadingError && (
+        <>{deletedNotes.length > 0 ? notesGrid : <p>Trash is empty</p>}</>
+      )}
+
+      <Modal show={showDeleteModal} onHide={() => setShowDeleteModal(false)}>
+        <Modal.Header closeButton>
+          <Modal.Title>Confirm Deletion</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          Are you sure you want to permanently delete this note?
+        </Modal.Body>
+        <Modal.Footer>
+          <Button variant="secondary" onClick={() => setShowDeleteModal(false)}>
```

Comment:
nit: if you have tertiary, use tertiary

Reply:
cancel can be the third option when you have 2 main actions on a modal, so better keep it as tertiary from the beginning